### ISSUE TITLE
Provide a way to specify default extension dependency in case the user won't provide any of required options

### ISSUE THREAD

**Issue by xstefank** [quarkusio/quarkus#28809]
### Description

We are getting the same problem with RR and RC options if we have an extension that can work with both but wants to allow user to pick their choice of either RR or RC when used with this extension:

https://github.com/quarkusio/quarkus/issues/24888
https://github.com/quarkusio/quarkus/issues/28744

It would be great if we could say in the extension (I'll use narayana-lra) the following:

* narayana-lra requires either resteasy-reactive-jackson or resteasy-jackson extension as a dependency
* if user specifies either or RR-jackson or RC-jackson use that
* If user don't specify any of the RR-jackson or RC-jackson use RR-jackson as automatic default dependency

In case of narayana this would be also the case rest-client extension so some generic approach would be preffered.

**Comment by xstefank** [comment]
@geoand, @ia3andy, @aloubyansky

**Comment by aloubyansky** [comment]
Feel free to revive the discussion we had before about it https://groups.google.com/g/quarkus-dev/c/08-edbpoeOQ/m/rZx5xRaIAQAJ?utm_medium=email&utm_source=footer

**Comment by aloubyansky** [comment]
Afaiu, #28744 has a bit of a different angle, i.e. it's more about project generation than dependency resolution.
Both issues aren't trivial. But the project generation one is probably what we should fix first.

**Comment by aloubyansky** [comment]
There is an issue and a discussion about it here https://github.com/quarkusio/quarkus/discussions/25757

**Comment by afalhambra** [comment]
Hello @aloubyansky, do we have any further update on this issue? would be nice to know the status of this.
Thank you!

**Comment by aloubyansky** [comment]
@afalhambra what specifically are you interested in? In injecting "default" dependencies during the dependency resolution or adding "default" dependencies to the POM/Gradle script when generating a new project?
There hasn't been any progress in any issue but it'd be good to clarify the demand :)

**Comment by aloubyansky** [comment]
Yes, this is what https://github.com/quarkusio/quarkus/issues/28744 is about.

**Comment by afalhambra** [comment]
Thanks @aloubyansky, appreciate it

**Comment by xstefank** [comment]
A few examples where this would be useful. Basically, any time we have multiple options for the same functionality. On top of my mind:

- LRA - either Quarkus REST or RESTEasy classic
- quarkus-logging-json (Quarkiverse) - Jackson or JSON-B
